crates/fuel-core/src/service/sub_services/algorithm_updater.rs

```diff
@@ -290,41 +301,47 @@ fn sync_metadata_storage_with_on_chain_storage(
     Ok(())
 }
 
-fn sync_v0_metadata(
+fn sync_v0_metadata<L2DataStore, L2DataStoreView>(
     settings: &ConsensusParametersProvider,
-    on_chain_db: Database<OnChain, RegularStage<OnChain>>,
+    on_chain_db: &L2DataStoreView,
     metadata_height: u32,
     latest_block_height: u32,
     updater: &mut AlgorithmUpdaterV0,
     metadata_storage: &mut StructuredStorage<
         Database<GasPriceDatabase, RegularStage<GasPriceDatabase>>,
     >,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<()>
+where
+    L2DataStore: L2Data,
+    L2DataStoreView: AtomicView<LatestView = L2DataStore>,
+{
     let first = metadata_height.saturating_add(1);
     let view = on_chain_db.latest_view()?;
     for height in first..=latest_block_height {
         let block = view
-            .storage::<FuelBlocks>()
-            .get(&height.into())?
-            .ok_or(anyhow::anyhow!("Expected block to exist"))?;
-        let last_tx_id = block.transactions().last().ok_or(anyhow::anyhow!(
-            "Expected block to have at least one transaction"
-        ))?;
+            .get_block(&height.into())?
+            .ok_or(not_found!("FullBlock"))?;
         let param_version = block.header().consensus_parameters_version;
```

Comment:
When `InitializeTask` will be moved into `gas-price-service`, instead of duplicating logic we can call `GasPriceTaskV0::process_l2_block(block)` inside of `InitializeTask::into_task`. Just a reminder to not forget

Reply:
yes, that's on the roadmap :)